lib/commands/skill/add-locales/ui.js

```diff
@@ -0,0 +1,56 @@
+const fs = require('fs');
+const inquirer = require('inquirer');
+const path = require('path');
+
+const Messenger = require('@src/view/messenger');
+
+module.exports = {
+    selectLocales,
+    displayAddLocalesResult
+};
+
+/**
+ * Prompt users to select locales
+ * @param {Array} localeChoices
+ * @param {Function} callback
+ */
+function selectLocales(localeChoices, callback) {
+    inquirer.prompt([{
+        message: 'Please select at least one locale to add:',
+        type: 'checkbox',
+        name: 'localeList',
+        choices: localeChoices
+    }]).then((answer) => {
+        callback(null, answer.localeList);
+    }).catch((error) => {
+        callback(error);
+    });
+}
+
+/**
+ * Display the result of locales addition
+ * @param {Array} selectedLocales selected locales
+ * @param {Map} iModelSourceByLocales Map { locale: filePath/remoteURI }
+ */
+function displayAddLocalesResult(selectedLocales, iModelSourceByLocales) {
+    for (const locale of selectedLocales) {
+        if (!iModelSourceByLocales.has(locale)) {
+            Messenger.getInstance().info(`Locale ${locale}.json already exists`);
+        }
+    }
+    Messenger.getInstance().info('');
+    if (iModelSourceByLocales.size === 0) {
+        return;
+    }
+
+    Messenger.getInstance().info('The following skill locale(s) have been added according to your local project:');
+    iModelSourceByLocales.forEach((v, k) => {
+        if (fs.existsSync(v)) {
```

Comment:
this feel hacky. sometimes "v" is file path and sometimes it is url. when you create iModelSourceByLocales you already do existsSync. "v" should be object with some flag { fromTemplate: true, sourceLocale: 'en', ...}

Reply:
Yes good point, I will avoid another fs existsync check, by passing an additional flag when resolving this Map